homeassistant/util/unit_system.py

```diff
@@ -84,14 +84,14 @@ def __init__(
         self,
         name: str,
         *,
-        accumulated_precipitation: str,
+        accumulated_precipitation: UnitOfPrecipitationDepth,
         conversions: dict[tuple[SensorDeviceClass | str | None, str | None], str],
-        length: str,
-        mass: str,
-        pressure: str,
-        temperature: str,
-        volume: str,
-        wind_speed: str,
+        length: UnitOfLength,
+        mass: UnitOfMass,
+        pressure: UnitOfPressure,
+        temperature: UnitOfTemperature,
```

Comment:
Alexa references `hass.config.units.temperature_unit` which was typed as string and cause mypy to complain.
I took the opportunity to adjust all `__init__` type hints.